pkg/urbit/jets/b/reel.c

```diff
@@ -3,34 +3,44 @@
 */
 #include "all.h"
 
-  static u3_noun
-  _reel_in(u3j_site* sit_u, u3_noun a, u3_noun b)
-  {
-    if ( 0 == a ) {
-      return b;
-    }
-    else if ( c3n == u3du(a) ) {
-      return u3m_bail(c3__exit);
-    }
-    else {
-      u3_noun gim = u3k(u3h(a));
-      u3_noun hur = _reel_in(sit_u, u3t(a), b);
-
-      return u3j_gate_slam(sit_u, u3nc(gim, hur));
-    }
-  }
-
 /* functions
 */
   u3_noun
   u3qb_reel(u3_noun a,
             u3_noun b)
   {
-    u3_noun  pro;
-    u3j_site sit_u;
-    u3j_gate_prep(&sit_u, u3k(b));
-    pro = _reel_in(&sit_u, a, u3k(u3x_at(u3x_sam_3, b)));
-    u3j_gate_lose(&sit_u);
+    u3_noun pro = u3k(u3x_at(u3x_sam_3, b));
+    if ( u3_nul != a ) {
+      u3j_site sit_u;
+      u3_noun  i, t = a;
+      c3_w     j_w, len_w = 0, all_w = 89, pre_w = 55;
+      u3_noun* vec = u3a_malloc(all_w * sizeof(u3_noun));
+
+      // stuff list into an array
+      do {
+        if ( c3n == u3r_cell(t, &i, &t) ) {
+          u3a_free(vec);
+          return u3m_bail(c3__exit);
+        }
```

Comment:
I think this can just be `u3x_cell()` -- we don't need to clean up our allocations if we bail. And this code already depends on that being the case, consider the refcount of `pro`.